matrix-sdk/src/main/java/org/matrix/androidsdk/data/store/MXMemoryStore.java

```diff
@@ -655,6 +663,16 @@ public void storeLiveRoomEvent(Event event) {
         }
     }
 
+    private boolean isLastEventInRightPlace(LinkedHashMap<String, Event> events) {
+        if (events.size() < 2) return true;
+        String[] eventKeys = (String[]) events.keySet().toArray();
```

Comment:
vu que tu fais un `events.get(key)` deux lignes plus bas, ça peut valoir le coup de ne pas juste récupérer les keys, mais les entries (la paire clé/valeur)
ton get il va ré-itérer sur toute la liste : pas top